pkg/api/v1/defaults.go

```diff
@@ -117,6 +119,42 @@ func SetDefaults_VirtualMachine(obj *VirtualMachine) {
 	if obj.Spec.Domain.Features == nil {
 		obj.Spec.Domain.Features = &Features{}
 	}
+	obj.Spec.Domain.Machine.Type = "q35"
+	setDefaults_DiskFromMachineType(obj)
+}
+
+func setDefaults_DiskFromMachineType(obj *VirtualMachine) {
+	bus := diskBusFromMachine(obj.Spec.Domain.Machine.Type)
+
+	for i := range obj.Spec.Domain.Devices.Disks {
+		disk := &obj.Spec.Domain.Devices.Disks[i].DiskDevice
+
+		SetDefaults_DiskDevice(disk)
+
+		if disk.Disk != nil && disk.Disk.Bus == "" {
+			disk.Disk.Bus = bus
+		}
+		if disk.CDRom != nil && disk.CDRom.Bus == "" {
+			disk.CDRom.Bus = bus
+		}
+		if disk.LUN != nil && disk.LUN.Bus == "" {
+			disk.LUN.Bus = bus
+		}
+		if disk.Floppy != nil && disk.Floppy.Bus == "" {
+			// no real choice here
+			disk.Floppy.Bus = "fd"
```

Comment:
Is that the only reasonable value for floppy in general?

Reply:
I'm not 100% sure that it is the only reasonable value for floppies, e.g we can attach one floppy through usb. But surely it is a supported value, and, looking at libvirt sources, seems the preferred one. And it should be "fdc" (see below)